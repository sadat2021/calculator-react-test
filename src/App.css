html {
  box-sizing: border-box;
}

*, *:before, *:after {
  box-sizing: inherit;
}

body {
  margin: 0;
  font: 100 14px "Roboto";
}

button {
  display: block;
  background: none;
  border: none;
  padding: 0;
  font-family: inherit;
  cursor: pointer;
  outline: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  touch-action: manipulation;
}

button:active {
  box-shadow: inset 0px 0px 80px 0px rgba(0, 0, 0, 0.25);
}

#wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
}

#calculator-wrapper {
  height: 560px;
  position: relative;
  border: 20px solid #1c191c;
  border-radius: 20px;
  box-shadow: 0px 0px 20px 0px #aaa;
}

.calculator {
  width: 100%;
  height: 100%;
  background: black;
  display: flex;
  flex-direction: column;
}

.calculator-display {
  color: white;
  background: #1c191c;
  line-height: 130px;
  font-size: 6em;
  flex: 1;
}

.auto-scaling-text {
  display: inline-block;
}

.calculator-display .auto-scaling-text {
  padding: 0 16px;
  position: absolute;
  right: 0;
  transform-origin: right;
}

.calculator-keypad {
  height: 400px;
  display: flex;
}

.calculator .input-keys {
  width: 240px;
}

.calculator .function-keys {
  display: flex;
}

.calculator .digit-keys {
  background: #1c191c;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap-reverse;
}

.calculator .scientific-keys {
  background: #1c191c;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 480px;
}

.key-scientific {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 16px!important;
}

.calculator .scientific-keys .calculator-key {
  font-size: 2.25em;
  color: #c3c3c3;
  background-color: #2f2c2c;
}

.calculator-key {
  width: 80px;
  height: 80px;
  text-align: center;
  line-height: 60px;
  border: 8px solid #1c191c;
  border-radius: 50%;
}

.calculator .function-keys .calculator-key {
  font-size: 1.6em;
}

.calculator .digit-keys .calculator-key {
  font-size: 2.25em;
  color: white;
  background-color: #474747;
}

.calculator .digit-keys .key-0 {
  width: 160px;
  text-align: left;
  padding-left: 26px;
  border-radius: 36px;
}

.calculator .digit-keys .key-dot {
  padding-top: 0.1em;
}



.calculator .function-keys, .calculator .operator-keys {
  background-color: #1c191c;
}

.calculator .function-keys .calculator-key {
  background-color: #b4b4b4;
}

.active-key {
  background-color: white;
  font-size: 2.2em;
  color: orange;
}

.inactive-key {
  color: white;
  font-size: 2.2em;
  background-color: orange;
}

/* Mobile phone */

@media (max-width: 768px) {
  #calculator-wrapper {
    width: 332px;
    height: 536px;
    position: relative;
    border: 8px solid #1c191c;
    border-radius: 8px;
    box-shadow: none;
  }
  .calculator .scientific-keys {
    display: none;
  }
}

/* Very small phone */

@media (max-width: 320px) {
  .App-title, .App-intro, .version {
    display: none;
  }
  #calculator-wrapper {
    height: 100vh;
  }
}