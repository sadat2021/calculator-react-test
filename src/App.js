import React from "react";
import Calculator from "./components/Calculator";
import "./App.css";

function App() {
  return (
    <div className="App">
      <div id="wrapper">
        <div id="calculator-wrapper">
          <Calculator />
        </div>
      </div>
    </div>
  );
}

export default App;
