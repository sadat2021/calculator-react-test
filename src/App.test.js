import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, mount, shallow } from "enzyme";
import App from "./App";

configure({ adapter: new Adapter() });

describe("Calculator React Test", () => {
  it("takes a snapshot from the Calculator component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
  it("checks to see add and equals opertion work correctly", () => {
    const wrapper = mount(<App />);

    const key1 = wrapper.find('[data-testid="key1"]');
    const key2 = wrapper.find('[data-testid="key2"]');
    const display = wrapper.find('[data-testid="display"]');
    const key5 = wrapper.find('[data-testid="key5"]');
    const key7 = wrapper.find('[data-testid="key7"]');
    const add = wrapper.find('[data-testid="add"]');
    const equals = wrapper.find('[data-testid="equals"]');

    expect(key1.length).toBe(1);
    expect(key2.length).toBe(1);
    expect(display.length).toBe(1);
    expect(key5.length).toBe(1);
    expect(key7.length).toBe(1);
    expect(add.length).toBe(1);
    expect(equals.length).toBe(1);

    key1.at(0).simulate("click");
    key2.at(0).simulate("click");
    add.at(0).simulate("click");
    key5.at(0).simulate("click");
    key7.at(0).simulate("click");
    equals.at(0).simulate("click");

    expect(display.at(0).text()).toBe("69");
  });
});
