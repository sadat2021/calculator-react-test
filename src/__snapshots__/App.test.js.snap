// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator React Test takes a snapshot from the Calculator component 1`] = `
<div
  className="App"
>
  <div
    id="wrapper"
  >
    <div
      id="calculator-wrapper"
    >
      <Calculator />
    </div>
  </div>
</div>
`;

exports[`card builder takes a snapshot from the HCardBuilder component 1`] = `
<div
  className="App"
>
  <div
    id="wrapper"
  >
    <div
      id="calculator-wrapper"
    >
      <Calculator />
    </div>
  </div>
</div>
`;
